test/e2e_node/eviction_test.go

```diff
@@ -277,6 +281,53 @@ var _ = framework.KubeDescribe("LocalStorageCapacityIsolationEviction [Slow] [Se
 	})
 })
 
+func runOneQuotaTest(f *framework.Framework, quotasRequested bool) {
+	evictionTestTimeout := 10 * time.Minute
+	sizeLimit := resource.MustParse("100Mi")
+	useOverLimit := 101 /* Mb */
+	useUnderLimit := 99 /* Mb */
+	priority := 0
+	if quotasRequested {
+		priority = 1
+	}
+	Context(fmt.Sprintf(testContextFmt, fmt.Sprintf("evictions due to pod local storage violations checked by quota (enabled: %v)", quotasRequested)), func() {
+		tempSetCurrentKubeletConfig(f, func(initialConfig *kubeletconfig.KubeletConfiguration) {
+			// TODO: remove hardcoded kubelet volume directory path
+			// framework.TestContext.KubeVolumeDir is currently not populated for node e2e
+			if quotasRequested && !supportsQuotas("/var/lib/kubelet") {
+				// No point in running this as a positive test if quotas are not
+				// enabled.
+				framework.Skipf("Cannot run LocalStorageCapacityIsolationQuotaEviction on non-quota-enabled environment")
+			}
+			// setting a threshold to 0% disables; non-empty map overrides default value (necessary due to omitempty)
+			initialConfig.EvictionHard = map[string]string{"memory.available": "0%"}
+			initialConfig.FeatureGates[string(features.LocalStorageCapacityIsolationFSQuotaMonitoring)] = quotasRequested
+		})
+		runEvictionTest(f, evictionTestTimeout, noPressure, noStarvedResource, logDiskMetrics, []podEvictSpec{
+			{
+				evictionPriority: priority, // This pod should be evicted because of emptyDir violation if quotas are enabled
+				pod: diskConcealingPod(fmt.Sprintf("emptydir-concealed-disk-sizelimit-%v", quotasRequested), useOverLimit, &v1.VolumeSource{
+					EmptyDir: &v1.EmptyDirVolumeSource{SizeLimit: &sizeLimit},
+				}, v1.ResourceRequirements{}),
+			},
+			{
+				evictionPriority: 0, // This pod should not be evicted because it uses less than its limit (test for quotas)
+				pod: diskConcealingPod(fmt.Sprintf("emptydir-concealed-disk-below-sizelimit-%v", quotasRequested), useUnderLimit, &v1.VolumeSource{
+					EmptyDir: &v1.EmptyDirVolumeSource{SizeLimit: &sizeLimit},
+				}, v1.ResourceRequirements{}),
+			},
+		})
+	})
+}
+
+// LocalStorageCapacityIsolationEviction tests that container and volume local storage limits are enforced through evictions
+var _ = framework.KubeDescribe("LocalStorageCapacityIsolationQuotaEviction [Slow] [Serial] [Disruptive] [Feature:LocalStorageCapacityIsolationQuota][NodeFeature:Eviction]", func() {
+	defer utilfeature.DefaultMutableFeatureGate.SetFromMap(map[string]bool{string(features.LocalStorageCapacityIsolationFSQuotaMonitoring): utilfeature.DefaultFeatureGate.Enabled(features.LocalStorageCapacityIsolationFSQuotaMonitoring)})
+	f := framework.NewDefaultFramework("localstorage-eviction-quota-test")
+	runOneQuotaTest(f, true)
+	runOneQuotaTest(f, false)
```

Comment:
This case is essentially just the LocalStorageCapacityIsolation test, right?  Can we just explicitly set `LocalStorageCapacityIsolationFSQuotaMonitoring` to false in that test so we can avoid adding the extra test case?

Reply:
No, it's not the same test -- it deletes the file while holding it open.  With quota monitoring turned off, the node should not be evicted.

Reply:
Can we merge the two tests?  I.E. the LocalStorageCapacityIsolation is the test with the feature gate disabled, and the new one is with it enabled?  Just add the pod which deletes the file while holding it open to the LSCI test.  Then eventually once it is on by default and the feature gate is removed, we merge the whole thing together into a single test.

Reply:
I think for now I'd rather keep these tests separate, at least for the time being.  The quota test is focused on testing that the quota mechanism works correctly; it uses eviction as a mechanism to detect that by running cases that distinguish between quotas enabled and disabled and that distinguish between being slightly over and slightly under the limit.  The LSCI test is focused on verifying that the eviction mechanism works correctly.

I was originally going to create an entirely new quota_test.go; there's just a lot of overlap of functionality between the two.

We can always re-evaluate this later, but even when the feature gate is removed, we have to handle the case where quotas are not enabled (the filesystem doesn't support quotas).

Reply:
That makes sense to me.  We don't want to skip the LSCI test if quotas are disabled.